### PR TITLE
Add YAMLLint to CI

### LINKED ISSUE #1688: [repo] Add linting of yml/yaml files in CI
We ran into an issue where a typo in a yml file (https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1672/commits/09cafa5c36537a7743bee9150ecb18babd6e6918) broke all code coverage upload.

I did a quick scan of GitHub marketplace, there seem to be some YML linters available. We should add linting of yml/yaml files into the CI.

High level we need to add a bucket for these files here: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/dc7be37862e4669638bb9df42db83794cbb667d7/.github/workflows/ci.yml#L20

`yml: [ '**.yml, '**.yaml' ]`

And then add a step for `yml` changes which calls some workflow to run linting. Similar to this...

https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/dc7be37862e4669638bb9df42db83794cbb667d7/.github/workflows/ci.yml#L78-L83

### PR DESCRIPTION
**Pull request by mowies**
Fixes #1688

## Changes

- added yamllint CLI to the CI pipeline to check all YAML files in the repo
- adapted contributing guidelines to reflect changes as well
- added default linter ruleset copied from https://github.com/open-telemetry/opentelemetry-demo/blob/main/.yamllint

#### Note
There's are lots of errors showing right now, using the default ruleset.
For now, I copied the ruleset from https://github.com/open-telemetry/opentelemetry-demo/blob/main/.yamllint which seems reasonable to me, but there's still some linting errors. I can definitely fix the remaining linter errors if somebody here tells me to :)

## Merge requirement checklist

* [x] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet-contrib/blob/main/CONTRIBUTING.md) guidelines followed (license requirements, nullable enabled, static analysis, etc.)
* [ ] Unit tests added/updated
* [ ] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Kielek) [inline .github/workflows/ci.yml:21]: I think that, you should add here also `.yamlint` file
```suggestion
          yml: [ '**.yml, '**.yaml', '.yamlint' ]
```

File: .github/workflows/ci.yml
Comment (Kielek) [inline .github/workflows/ci.yml:21]:
```suggestion
          yml: [ '**.yml', '**.yaml', '.yamllint' ]
```

Comment (mowies) [reply .github/workflows/ci.yml:21]: woops 😅

**Comment by Kielek** [comment]
@mowies, thanks for doing this.

As you described, this job is failing: https://github.com/open-telemetry/opentelemetry-dotnet-contrib/actions/runs/9739790217/job/26875671492?pr=1933.
Before merging we need to fix it.
I have a couple of comments/questions there

1. Lister reports both Errors and Warnings. What happen if it will have only warnings? Will it fail or just left a comment? If it only left a comment in the execution, we should either disabled it, or better treat as errors. We do not need to soft recommendation. Nobody will be checking the CI output.
2. Line length - I think that we could just disabled this/double the value to 400. I do not see this as problematic (just personal opinion, you can fix all occurrences to 200).
3. Please fix all other, obvious issues reporter by the linter.

**Comment by mowies** [comment]
> 1. Lister reports both Errors and Warnings. What happen if it will have only warnings? Will it fail or just left a comment? If it only left a comment in the execution, we should either disabled it, or better treat as errors. We do not need to soft recommendation. Nobody will be checking the CI output.

So we have 2 options for this: strict mode and ignoring warnings
- strict mode: yamllint will treat warnings as errors and will return with a non-zero exit code also when there's just warnings
- ignoring warnings: yamllint will not print any warnings, just errors

> 2. Line length - I think that we could just disabled this/double the value to 400. I do not see this as problematic (just personal opinion, you can fix all occurrences to 200).

I will disable the rule.

**Comment by mowies** [comment]
@Kielek fixed all current linter errors, ignored the warnings for now. Please see my previous comment :)

**Comment by Kielek** [comment]
@movies, I think that we can ignore all warnings in this PR and enable strict mode in follow up/disable unwanted warnings one by one.

Now it pulutes output and PRs:
image
image

BTW, it will be great if you allow maintainers to modify your pr-branch. With this, we can merge all changes from current main, and merge it without waiting for you.

**Comment by mowies** [comment]
> @movies, I think that we can ignore all warnings in this PR and enable strict mode in follow up/disable unwanted warnings one by one.

Sounds good @Kielek , I changed the command to include `--no-warnings` and updated the contrib guide accordingly.

> BTW, it will be great if you allow maintainers to modify your pr-branch. With this, we can merge all changes from current main, and merge it without waiting for you.

It seems that I can't, since my fork is owned by an organization and not my personal account...
But I merged main now.

**Review by Kielek — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by vishweshbankwar — APPROVED** [review]

**Comment by Kielek** [comment]
@mowies, could you please merge main branch to your PR branch? I think that we are good to merge.

Consider creating follow up issues/PRs with enabling more strict validation.

**Pull request merged**
